src/lib.rs

```diff
@@ -175,6 +215,56 @@ impl Client {
         self.fetch::<Hotspot>(&format!("/hotspots/{}", address))
     }
 
+    /// Get details for a given hotspot address
+    pub fn get_hotspot_rewards(
+        &self,
+        address: &str,
+        min_time: Option<&str>,
+        max_time: Option<&str>,
+        cursor: Option<&str>,
+    ) -> error::Result<(Vec<Reward>, Option<String>)> {
+        let base = Url::parse(&format!("{}/", self.base_url)).unwrap();
+        let mut request_url = base.join(&format!("hotspots/{}/rewards", address)).unwrap();
+        for (k, v) in [
+            ("min_time", min_time),
+            ("max_time", max_time),
+            ("cursor", cursor),
+        ]
+        .iter()
+        {
+            if v.is_some() {
+                request_url.query_pairs_mut().append_pair(k, v.unwrap());
```

Comment:
See above.. I think parse_with_params may be a cleaner solution. If not, at least hang on to the results of query_pairs_mut so you can call append_pair without re-requesting the mutable query pairs every time?

Reply:
This now uses the `build_url` helper